tests/unit/FillInEntriesTest.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace viget\base\tests;
+
+use Codeception\Test\Unit;
+
+use UnitTester;
+use Craft;
+use craft\elements\Entry;
+
+use viget\base\Module;
+use viget\base\tests\fixtures\EntriesFixture;
+
+class FillInEntriesTest extends Unit
+{
+    /**
+     * @var UnitTester
+     */
+    protected $tester;
+
+    public $entryIds = [];
+    public $params = [
+        'section' => 'testSection1',
+    ];
+
+    public function _fixtures(): array
+    {
+        return [
+            'entries' => [
+                'class' => EntriesFixture::class,
+            ],
+        ];
+    }
+
+    protected function _before()
+    {
+        $this->eventIds = Entry::find()->section('testSection1')->ids();
+    }
+
+    public function testNotEnough()
+    {
+        $selectedEntries = Entry::findAll([
+            'section' => 'testSection1',
+            'limit' => 2,
+        ]);
+
+        $entries = Module::$instance->util->fillInEntries($selectedEntries, $this->params, 4);
+
+        $this->assertEquals(count($entries), 4);
+    }
+
+    public function testEnoughPassed()
+    {
+        $selectedEntries = Entry::findAll([
+            'section' => 'testSection1',
+            'limit' => 4,
+        ]);
+
+        $entries = Module::$instance->util->fillInEntries($selectedEntries, $this->params, 4);
+
+        $this->assertEquals(count($entries), 4);
+    }
```

Comment:
Should we add a "more than enough" test?

Reply:
Hmm not sure I follow

Reply:
like

```php
$selectedEntries = Entry::findAll([
    'section' => 'testSection1',
    'limit' => 5,
]);
$entries = Module::$instance->util->fillInEntries($selectedEntries, $this->params, 4);
$this->assertEquals(count($entries), 5);
```

Maybe not necessary 🤷‍♂️

Reply:
re: your other comment and this one, is it better to just change the behavior to respect the limit no matter what?

Reply:
sure and then only use `fillInEntries` if you know need more? makes sense! I think either way works